### DISCUSSION TITLE
Optimizing Adaptive Prompt Design for Anthropic API's Dynamic Context Cache

### DISCUSSION THREAD

**Discussion by mkXultra** [yihong0618/bilingual_book_maker#424] [Ideas]
# Optimizing Adaptive Prompt Design for Anthropic API's Dynamic Context Cache

## Background
Anthropic has recently implemented a Context Cache feature in their API. This new functionality has the potential to significantly reduce costs and potentially outperform GPT-4 in certain scenarios. However, due to the dynamic nature of the Context Cache, careful consideration must be given to prompt design to fully leverage its benefits.

https://www.anthropic.com/pricing#anthropic-api
https://www.anthropic.com/news/prompt-caching

**Comment by yihong0618** [comment]
agree, just go for it
Btw, thanks a lot for your work these days, and sorry for I did not add new features for a bit time~

**Reply by jossef123456789** [reply]
Hi your project is amazing but there is a really great feature i think you should add and it is to support all openrouter models really they are so great there free ones can really do the job without paying any money and yeah also i wish you add a file an empty one when the user should add in just every line the ai api key and the program will take that api key every time from that file it will be really so much better for the user and for the program also in case i want to watch a 1000 book like war and peace by leo it will require so many api keys and for that putting them in a file that the program will use it will be much better it will be even automated by making a script to put them so yeah just remember the openrouter support and the file that contain the api keys and thank you for this great project

**Reply by yihong0618** [reply]
thank you for the words will try to support it later maybe
and Pull request welcome
